syntax = "proto3";
option cc_generic_services = true;
option java_generic_services = true;

import "common.proto";

package calendar; 

message CalendarItem
{
   common.UniqueId id = 1;
   string text = 2;
   int64 timestamp = 3;
}

message CalendarItems
{
   repeated CalendarItem items = 1;
}

message NextQueryParameters
{
   int64 timestampFrom = 1;
   int32 limit = 2;
}

service CalendarService
{
   rpc Query (common.TimeRange) returns (CalendarItems);
   rpc Next (NextQueryParameters) returns (CalendarItems);
   rpc DeleteItem (common.UniqueId) returns (common.OperationResultMessage);
   rpc EditItem (CalendarItem) returns (common.OperationResultMessage);
   rpc AddItem (CalendarItem) returns (common.UniqueId);
   rpc Clear(common.EmptyMessage) returns (common.OperationResultMessage);
}

